### ISSUE TITLE
fix(ml): keep BatchNorm layers in inference mode during phase 2 fine-tuning

### ISSUE THREAD

**Issue by LukeSantossz** [LukeSantossz/visiosoil-app#179]
## Description

Phase 2 fine-tuning unfreezes the top backbone layers without returning their BatchNormalization layers to inference mode. Those layers then overwrite the ImageNet running statistics with statistics estimated from a very small, single-rig dataset, degrading the model silently.

## Context

Found by the code audit of 2026-08-22 against `main` at 4635804.

This is the standard transfer-learning trap and Keras' own transfer-learning guidance addresses it directly: when a pretrained backbone is partially unfrozen, its BatchNormalization layers must be kept in inference mode, or fine-tuning destroys the statistics that pretraining produced.

It matters more here than in a typical fine-tune. The dataset is photographed on one rig, with one device and one lighting condition (`docs/ml/collection-protocol.md:33-38`), so the batch statistics the layers would learn describe a single capture configuration rather than the population the model is deployed against.

## Current State

`ml/src/model.py:102-106`:

```python
# Unfreeze the backbone
backbone.trainable = True

# Freeze all layers except the last `unfreeze_layers`
for layer in backbone.layers[:-unfreeze_layers]:
    layer.trainable = False
```

With the configured `model.unfreeze_layers: 50` (`ml/config.yaml`), 17 BatchNormalization layers sit inside the unfrozen tail and are therefore trainable. A trainable BN layer runs in training mode during `fit` and updates its moving mean and variance, at `training.batch_size: 32`.

No test covers this. `ml/tests/test_model_output.py:124-145` asserts only that the trainable tail is non-empty.

## Proposed Solution

After computing which layers to unfreeze, explicitly set every BatchNormalization layer in the backbone back to non-trainable, and assert it.

```python
for layer in backbone.layers:
    if isinstance(layer, keras.layers.BatchNormalization):
        layer.trainable = False
```

Criteria for this shape: it is the documented Keras remedy, it is three lines, it does not change the unfreeze count the config declares, and it is directly assertable in a test.

The alternative — leaving BN trainable and lowering the fine-tune learning rate — does not work, because the running statistics are updated by the forward pass rather than by the optimizer, so the learning rate does not govern them.

## Acceptance Criteria

- [ ] `unfreeze_model` leaves every BatchNormalization layer in the backbone non-trainable.
- [ ] A test asserts that after `unfreeze_model`, no BatchNormalization layer in the backbone is trainable, and that the count of trainable non-BN layers still matches `model.unfreeze_layers`. It must fail on the current code.
- [ ] The trainable parameter count after unfreezing is recorded in `metrics.json`, so a future change to this logic is visible in the artifact rather than only in the code.
